crates/iceberg/src/puffin/compression.rs

```diff
@@ -15,9 +15,12 @@
 // specific language governing permissions and limitations
 // under the License.
 
+use serde::{Deserialize, Serialize};
+
 use crate::{Error, ErrorKind, Result};
 
-#[derive(Debug, PartialEq, Eq, Clone, Copy, Default)]
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Default, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
 /// Data compression formats
```

Comment:
I usually prefer to keep comments at the top before `#[derive]` to make them easier to find. This also provides more space for writing lengthy comments, which may include examples.

```rust
/// Data compression formats
#[derive(Debug, PartialEq, Eq, Clone, Copy, Default, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
```